scm/github.go

```diff
@@ -689,6 +689,42 @@ func (s *GithubSCM) GetUserScopes(ctx context.Context) *Authorization {
 	return &Authorization{Scopes: gitScopes}
 }
 
+// GetRepositoryInvites implements the SCM interface
+func (s *GithubSCM) AcceptRepositoryInvites(ctx context.Context, opt *RepositoryInvitationOptions) error {
+	if !opt.valid() {
+		return ErrMissingFields{
+			Method:  "GetRepositoryInvites",
+			Message: fmt.Sprintf("%+v", opt),
+		}
+	}
+
+	invites, _, err := s.client.Users.ListInvitations(ctx, &github.ListOptions{})
+	if err != nil {
+		return ErrFailedSCM{
+			GitError: fmt.Errorf("failed to fetch GitHub repository invitations: %w", err),
+			Method:   "GetRepositoryInvites",
+			Message:  "failed to fetch GitHub repository invitations",
+		}
+	}
+
+	for _, invite := range invites {
+		if invite.Repo.Owner.GetLogin() != opt.Owner || invite.Invitee.GetLogin() != opt.Login {
+			// Ignore unrelated invites
+			continue
+		}
+
+		_, err := s.client.Users.AcceptInvitation(ctx, invite.GetID())
+		if err != nil {
+			return ErrFailedSCM{
```

Comment:
This error could now report which specific repository for which the invitation failed; I guess that could be useful to at least log in the system log.